src/engine/strat_engine/backstore/device.rs

```diff
@@ -43,3 +46,159 @@ pub fn resolve_devices<'a>(paths: &'a [&Path]) -> StratisResult<HashMap<Device,
     }
     Ok(map)
 }
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum DevOwnership {
+    Ours(PoolUuid, DevUuid),
+    Unowned,
+    Theirs(String), // String is something useful to give back to end user about what's on device
+}
+
+/// Returns true if a device has no signature, yes this is a bit convoluted.  Logic gleaned from
+/// blivet library.
+fn empty(device: &HashMap<String, String>) -> bool {
+    if (device.contains_key("ID_PART_TABLE_TYPE") && !device.contains_key("ID_PART_ENTRY_DISK"))
+        || device.contains_key("ID_FS_USAGE")
+    {
+        false
+    } else {
+        true
+    }
+}
```

Comment:
another option:
```
    !((device.contains_key("ID_PART_TABLE_TYPE") && !device.contains_key("ID_PART_ENTRY_DISK"))
        || device.contains_key("ID_FS_USAGE"))
```
might be better? OR go the other way with:
```
    if device.contains_key("ID_PART_TABLE_TYPE") {
        if device.contains_key("ID_PART_ENTRY_DISK") {
            true
        } else {
            false
        }
    } else if device.contains_key("ID_FS_USAGE") {
        false
    } else {
        true
    }
```
might actually be best. Doing a little of both seems odd, though.

Reply:
I prefer first suggested option.

Reply:
Done